sympy/ntheory/residue_ntheory.py

```diff
@@ -1300,3 +1300,52 @@ def discrete_log(n, a, b, order=None, prime_order=None):
         return _discrete_log_pollard_rho(n, a, b, order)
 
     return _discrete_log_pohlig_hellman(n, a, b, order)
+
+
+
+def quadratic_congruence(a, b, c, p):
+    """
+    Find the solutions to ``a x**2 + b x + c = 0 mod p
+    a : integer
+    b : integer
+    c : integer
+    p : positive integer
+    """
+    from sympy.polys.galoistools import linear_congruence
+    a = as_int(a)
+    b = as_int(b)
+    c = as_int(c)
+    p = as_int(p)
+    a = a % p
+    b = b % p
+    c = (c % p)
+
+    if a == 0:
+        return linear_congruence(b, -1 * c, p)
+    if p == 2:
+        roots = set()
+        if c % 2 == 0:
+            roots.add(0)
+        if (a + b + c) % 2 == 0:
+            roots.add(1)
+        return roots
```

Comment:
A test for the case when 0 and 1 are returned should be added and a case for when only 1 is returned.